### PR TITLE
Add a method to RpcModule that transforms the module into a `RpcModule<()>`, i.e. removes the context.

### LINKED ISSUE #558: Make it trivial to convert a `RpcModule<Something>` to `RpcModule<()>`
Atm users sometimes have to do this to get an `RpcModule` with a void context, `()`:

```rust
		let rpc_api = SomeRpcThing::new(ctx).into_rpc();
		let mut module = RpcModule::new(());
		module.merge(rpc_api).expect("only one module; qed");
		Ok(module)
```

Add a QoL method to `RpcModule` that consumes `self` and returns a `RpcModule<()>`.

### PR DESCRIPTION
**Pull request by dvdplm**
Closes #558﻿

### CODE REVIEW

File: core/src/server/rpc_module.rs [hunk 1]
```diff
@@ -504,6 +504,13 @@ impl<Context> RpcModule<Context> {
 	pub fn new(ctx: Context) -> Self {
 		Self { ctx: Arc::new(ctx), methods: Default::default() }
 	}
+
+	/// Transform a module into an `RpcModule<()>` (unit context).
+	pub fn decontextualize(self) -> Result<RpcModule<()>, Error> {
+		let mut module = RpcModule::new(());
+		module.merge(self)?;
```
Comment (niklasad1) [inline core/src/server/rpc_module.rs:511]: this could never fail right, as you merge with an empty RpcModule?

Comment (dvdplm) [reply core/src/server/rpc_module.rs:511]: 🤦

File: core/src/server/rpc_module.rs [hunk 2]
```diff
@@ -504,6 +504,13 @@ impl<Context> RpcModule<Context> {
 	pub fn new(ctx: Context) -> Self {
 		Self { ctx: Arc::new(ctx), methods: Default::default() }
 	}
+
+	/// Transform a module into an `RpcModule<()>` (unit context).
+	pub fn decontextualize(self) -> RpcModule<()> {
```
Comment (niklasad1) [inline core/src/server/rpc_module.rs:509]: I found the method name a bit long to read, could `strip_context`, `remove_context` or something work?

**Review by niklasad1 — APPROVED** [review]

File: tests/tests/rpc_module.rs [hunk 3]
```diff
@@ -31,6 +31,23 @@ use jsonrpsee::core::Error;
 use jsonrpsee::types::{EmptyParams, Params};
 use serde::{Deserialize, Serialize};
 
+fn assert_type<T: 'static, Expected: 'static>(_expected: &Expected, explain: Option<&'static str>) {
+	use core::any::TypeId;
+	match explain {
+		None => assert_eq!(TypeId::of::<T>(), TypeId::of::<Expected>()),
+		Some(explain) => assert_eq!(TypeId::of::<T>(), TypeId::of::<Expected>(), "{}", explain),
+	}
+}
```
Comment (jsdw) [inline tests/tests/rpc_module.rs:40]: Perhaps it's better to move this check to compile-time? ie:

```rust
fn assert_type<Expected: 'static>(_expected: &Expected)
```

Usage (compile error if types don't match):

```
assert_type::<SomeType>(some_value);
```

Comment (dvdplm) [reply tests/tests/rpc_module.rs:40]: cheers!

**Review by jsdw — APPROVED** [review]

File: core/src/server/rpc_module.rs [hunk 4]
```diff
@@ -504,6 +504,13 @@ impl<Context> RpcModule<Context> {
 	pub fn new(ctx: Context) -> Self {
 		Self { ctx: Arc::new(ctx), methods: Default::default() }
 	}
+
+	/// Transform a module into an `RpcModule<()>` (unit context).
+	pub fn remove_context(self) -> RpcModule<()> {
+		let mut module = RpcModule::new(());
+		module.merge(self).expect("Merging with an empty module can never have conflicting method names.");
```
Comment (maciejhirsz) [inline core/src/server/rpc_module.rs:511]:
```suggestion
		module.methods = self.methods;
```

Simpler, faster, and no need for `expect`.

**Review by maciejhirsz — APPROVED** [review]
Just one simple grumble :)

**Pull request merged**